### PR TITLE
UnionFind function

### PR DESCRIPTION
**Pull request by mhavrylova**
added UnionFind functionality with methods: 
 - connect;
 - isConnected;
 - count;
 - component.
Added test for this function

### CODE REVIEW

File: unionFind.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	isConnected(a, b) { 
+		if( isNaN(this._elements[a]) || isNaN(this._elements[b]) ) {
+			return false;
+		}
+		return this._elements[a] === this._elements[b];
+	}
+
+	count() {
+	    let result = [];
+	    nextInput:
```
Comment (pavlom0x) [inline unionFind.js:34]: `nextInput:` what is it?

**Comment by pavlom0x** [comment]
all looks good except count method, I don't understand it

File: unionFind.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		      		let str = this._elements[i];
+			      	for (let j = 0; j < result.length; j++) {
+			        	if (result[j] == str) continue nextInput;
+			      	}
+	 				if (!isNaN(str)) {
+	           			result.push(str);
+	          		}
+		    }
+		return result.length;
+	}
```
Comment (pavlom0x) [inline unionFind.js:45]: A little hard to read this method

File: unionFind.js (hunk 1 above)
Comment (mhavrylova) [reply unionFind.js:34]: this is label for exit from all levels of loop. Please see here: https://learn.javascript.ru/while-for#метки-для-break-continue

File: unionFind.js (hunk 2 above)
Comment (mhavrylova) [reply unionFind.js:45]: At first we create empty array "result" and then push there all unique elements from array "_elements". So the length of array "result" == count of our components.

**Pull request merged**